docker/Dockerfile

```diff
@@ -0,0 +1,37 @@
+ARG ROS_VERSION=jazzy
```

Comment:
Maybe also an ARG for the RMW_IMPLEMENTATION and ROS_DOMAIN_ID to use? Otherwise the container might have trouble connecting. Also an argument to choose between Rosbridge and RWS. Rosbridge as it currently stands is only really passable on a decent workstation with good single core speed, so for any kind of embedded usage it would make sense to have a RWS container too. Both need git to work on Jazzy right now anyhow.

Reply:
Breaking this up:

- [ ] Container currently connects fine to the host via shared memory. This volume map does the connection: `-v /dev/shm:/dev/shm ` Is the ROS_DOMAIN_ID something we should still chase?
- [ ] I'm new to RWS and RMW. It seems they are not mutually exclusive? When trying to implement the RWS only (requires CycloneDDS install) and launching, the launch file returns: `Use rmw_fastrtps_dynamic_cpp instead of rmw_fastrtps_cpp as RMW implementation.`

Reply:
If you have time, to test this build the container from the latest commit using: `docker build -f docker/Dockerfile -t vizanti:2.0 . --build-arg ROS_VERSION=jazzy --build-arg USE_RWS=true`

And test run: `docker run --rm -it --net=host --name vizanti-ros2 -v /dev/shm:/dev/shm vizanti:2.0`

Reply:
> I'm new to RWS and RMW. It seems they are not mutually exclusive? When trying to implement the RWS only (requires CycloneDDS install) and launching, the launch file returns: Use rmw_fastrtps_dynamic_cpp instead of rmw_fastrtps_cpp as RMW implementation.

Yep that's [because of the topic introspection that RWS needs to run more efficiently](https://github.com/v-kiniv/rws/issues/7#issuecomment-1870563692). Cyclone has it by default, and FastDDS doesn't, however they do have that dynamic build that adds that which works.

> -v /dev/shm:/dev/shm

I'm not familiar with this approach, how does one pass env vars though that? Tested it briefly today on humble, and it doesn't seem that the RMW_IMPLEMENTATION nor ROS_DOMAIN_ID get picked up automatically (I thought maybe the terminal in which docker run executes might do something in that regard but doesn't seem to). 

Otherwise it seems like a step in the right direction, RWS seems to build ok even if it doesn't run yet given that the RMW settings aren't right.

Reply:
Ok, I have these cleaned up. Check the latest push and comment above. ROS_DOMAIN_ID is the last open item.